packages/text-input/src/TextInput.spec.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import {
+  render,
+  cleanup,
+  fireEvent,
+  getByTitle,
+  getByText,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import TextInput, { State } from './TextInput';
+import { typeIs } from '@leafygreen-ui/lib';
+
+afterAll(cleanup);
+
+describe('packages/text-input', () => {
+  const className = 'test-text-input-class';
+  const label = 'Test Input Label';
+  const description = 'This is the description';
+  const error = 'This is the error message';
+  const validEmail = 'test.email@mongodb.com';
+  const invalidEmail = 'invalid.email';
+  const onChange = jest.fn();
+  onChange.mockReturnValue('none');
+
+  const renderedTextInput = render(
+    <TextInput
+      className={className}
+      label={label}
+      description={description}
+      onChange={onChange}
+      state={State.none}
+      optional={true}
+    />,
+  );
```

Comment:
I have a trick that'll make this a bit neater!

Instead of:
```js
  const className = 'test-text-input-class';
  const label = 'Test Input Label';
  const description = 'This is the description';
  const error = 'This is the error message';
  const validEmail = 'test.email@mongodb.com';
  const invalidEmail = 'invalid.email';
  const onChange = jest.fn();
  onChange.mockReturnValue('none');

  const renderedTextInput = render(
    <TextInput
      className={className}
      label={label}
      description={description}
      onChange={onChange}
      state={State.none}
      optional={true}
    />,
  );
```

try this:
```js
  const error = 'This is the error message';
  const validEmail = 'test.email@mongodb.com';
  const invalidEmail = 'invalid.email';
  const props = {
    className: 'test-text-input-class',
    label: 'Test Input Label',
    description: 'This is the description',
    onChange: jest.fn(),
    state: State.none,
    optional: true,
  };
 
  props.onChange.mockReturnValue('none');

  const renderedTextInput = render(<TextInput {...props} />);
```

Reply:
Much prettier. Thanks!